doc/source/cluster/kubernetes/k8s-ecosystem/istio.md

```diff
@@ -0,0 +1,319 @@
+(kuberay-istio)=
+# Enable mTLS and L7 observability of traffic in RayCluster with Istio
+
+This guide demonstrates the integration of KubeRay and Istio for enabling mTLS and L7 observability of traffic inside a RayCluster on a local Kind cluster.
+
+## Istio
+
+[Istio](https://istio.io/) is an open source service mesh that provides a uniform and more efficient way to secure, connect, and monitor services. Its powerful control plane brings vital features, including:
+* Secure network traffic in a Kubernetes cluster with TLS encryption.
+* Automatic metrics, logs, and traces for all traffic within a cluster.
+
+See the [Istio documentation](https://istio.io/latest/docs/) to learn more.
+
+## Step 0: Create a Kind cluster
+
+```bash
+kind create cluster
+```
+
+## Step 1: Install Istio
+
+```bash
+# Download istioctl and manifests
+export ISTIO_VERSION=1.21.1
+curl -L https://istio.io/downloadIstio | sh -
+cd istio-1.21.1
+export PATH=$PWD/bin:$PATH
+
+# Install Istio with:
+#   1. 100% trace sampling for demo purpose
+#   2. "sanitize_te" disabled for proper grpc interception. This is required by Istio 1.21.0 (https://github.com/istio/istio/issues/49685)
+#   3. TLS 1.3 enabled
+istioctl install -y -f - <<EOF
+apiVersion: install.istio.io/v1alpha1
+kind: IstioOperator
+spec:
+  meshConfig:
+    defaultConfig:
+      tracing:
+        sampling: 100
+      runtimeValues:
+        envoy.reloadable_features.sanitize_te: "false"
+    meshMTLS:
+      minProtocolVersion: TLSV1_3
+EOF
+
+# Install Istio addons, including kiali dashboard and jaeger dashboard
+kubectl apply -f samples/addons
+# Enable istio sidecar auto injection
+kubectl label namespace default istio-injection=enabled
+```
+
+See [Istio Getting Started](https://istio.io/latest/docs/setup/getting-started/) for more details on installing Istio.
+
+## Step 2: Install the KubeRay operator
+
+Follow [Deploy a KubeRay operator](kuberay-operator-deploy) to install the latest stable KubeRay operator from the Helm repository.
+
+## Step 3: Optional: Enable Istio mTLS STRICT mode
+
+This is an optional step to enable Istio mTLS in STRICT mode which provides the best security of service mesh by rejecting all undefined traffic.
+
+In this mode, we must disable the KubeRay init container injection by setting `ENABLE_INIT_CONTAINER_INJECTION=false` on the KubeRay controller. This is necessary because the init container starts before the istio-proxy, resulting in the rejection of all its network traffic in STRICT mode.
```

Comment:
```suggestion
In this mode, you must disable the KubeRay init container injection by setting `ENABLE_INIT_CONTAINER_INJECTION=false` on the KubeRay controller. This setting is necessary because the init container starts before the `istio-proxy`, resulting in the rejection of all of its network traffic in STRICT mode.
```